src/vs/workbench/contrib/notebook/browser/contrib/notebookVariables/notebookVariablesTree.ts

```diff
@@ -0,0 +1,64 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as dom from 'vs/base/browser/dom';
+import { IListVirtualDelegate } from 'vs/base/browser/ui/list/list';
+import { IListAccessibilityProvider } from 'vs/base/browser/ui/list/listWidget';
+import { ITreeNode, ITreeRenderer } from 'vs/base/browser/ui/tree/tree';
+import { FuzzyScore } from 'vs/base/common/filters';
+import { localize } from 'vs/nls';
+import { WorkbenchObjectTree } from 'vs/platform/list/browser/listService';
+
+export interface INotebookVariableElement {
+	readonly id: string;
+	readonly label: string;
+	readonly value: string;
+}
+
+export class NotebookVariablesTree extends WorkbenchObjectTree<INotebookVariableElement> { }
```

Comment:
No reason to subclass it, I think

Reply:
oh yeah, made more sense for the Async one with three arguments.